### PR TITLE
i18n: add args keyword to merge_file

### LINKED ISSUE #5293: i18n: allow to use keywords in i18n.merge_file()
It needed for libpeas-based plugin systems and another plugin systems, which relies on keyfile-based configs.

### PR DESCRIPTION
**Pull request by rilian-la-te**
I want to propose new version of #1630, because it is really useful. For example, for writing mate-panel applets and other systems, which relies on custom desktop files.
Also it allows to effectively translate ```appdata.xml.in``` files by passed ```-Lappdata``` to msgfmt args.

Fixes #5293

### CODE REVIEW

File: mesonbuild/modules/i18n.py [hunk 1]
```diff
@@ -85,6 +86,10 @@ def merge_file(self, state, args, kwargs):
         ]
         if datadirs:
             command.append(datadirs)
+
```
Comment (ignatenkobrain) [inline mesonbuild/modules/i18n.py]: __[Flake8]__ [W293] blank line contains whitespace ([link](https://sider.review/gh/repos/19784232/pulls/5300#issue-3997343))

<sub>posted by Sider</sub>

Comment (rilian-la-te) [reply mesonbuild/modules/i18n.py]: Fixed and force-pushed a fix.

**Comment by jpakkane** [comment]
Looks good, but this should have a test of some sort. Maybe edit one of the existing ones to take arguments?

**Comment by rilian-la-te** [comment]
@jpakkane thanks. Can I add test with second commit?

**Comment by jpakkane** [comment]
Sure.

**Comment by rilian-la-te** [comment]
@jpakkane do you need something more?

**Comment by jpakkane** [comment]
This is all that is needed. Thanks.

**Pull request merged**